src/validation.cpp

```diff
@@ -5066,7 +5066,7 @@ double GuessVerificationProgress(const ChainTxData& data, const CBlockIndex *pin
         fTxTotal = pindex->nChainTx + (nNow - pindex->GetBlockTime()) * data.dTxRate;
     }
 
-    return pindex->nChainTx / fTxTotal;
+    return std::min<double>(pindex->nChainTx / fTxTotal, 1.0);
```

Comment:
A comment explaining the truncate would be nice.

Reply:
An alternative is to truncate where it's needed, in L5064:
```
fTxTotal = std::max<double>(data.nTxCount + (nNow - data.nTime) * data.dTxRate, pindex->nChainTx);
```

Reply:
IMO this is fine as-is

- `std::min(..., bound)`  is a common enough way to cap values, it doesn't need explaining on every usage site
- it makes sense for "progress" values to never be able to get above 1.0 or below 0.0